private/registry-names-update.sps

```diff
@@ -0,0 +1,49 @@
+;; Copyright 2019 Lassi Kortela
+;; SPDX-License-Identifier: MIT
+
+(import (chezscheme))
+
+(define output-file "registry-names.sls")
+
+(define download-command
+  (string-append
+   "curl --fail --silent --show-error "
+   "https://raw.githubusercontent.com/"
+   "scheme-requests-for-implementation/srfi-common"
+   "/master/"
+   "admin/srfi-data.scm"))
+
+(define transcoder (make-transcoder (utf-8-codec) (eol-style lf)))
+
+(define (read-all port)
+  (let loop ((xs '()))
+    (let ((x (read port)))
+      (if (eof-object? x) (reverse xs) (loop (cons x xs))))))
+
+(define provided
+  (call-with-port (open-input-file "registry-provided.scm") read-all))
```

Comment:
Wouldn't this be better to be `iota` till some end number? This way even rejected/abandoned SRFI's gets listed. I have some SRFI's in IronScheme like that.

Reply:
@arcfide what is the opinion/policy of having non-accepted SRFI's ?

Reply:
The data in the srfi-common repo covers all SRFIs ever published. The present PR filters out from that list all such SRFIs that have not been included in chez-srfi. If it doesn't hurt for chez-srfi's name list to include the extra SRFIs not (yet) in chez-srfi, then we can simply remove the filter expression.

Reply:
Sometimes Scheme implementations still rely on or support withdrawn SRFIs, but it's not common. One problem with banishing withdrawn SRFIs from chez-srfi is that an SRFI's original author can withdraw it at any time, even years after it has been published. Mirroring that policy in chez-srfi would mean that working code can in principle be pulled from chez-srfi at any time so it supports less SRFIs than it did before. Maybe not ideal for users.

Reply:
@leppie:
> Wouldn't this be better to be `iota` till some end number? This way even rejected/abandoned SRFI's gets listed. I have some SRFI's in IronScheme like that.

If you include some withdrawn SRFIs in IronScheme that are not from chez-srfi, you'd be better off adding your own script in IronScheme to scrape https://raw.githubusercontent.com/scheme-requests-for-implementation/srfi-common/master/admin/srfi-data.scm directly.

P.S. Happy 2020 :)

Reply:
Hrm, I'm not sure I have a strong opinion here, as I never used the registry in the first place. However, I think the correct answer depends on how the registry is intended to be used. 

1. If the registry is meant to indicate what SRFIs the Chez-SRFIs project provides as a runtime introspectable value that can be used to, for instance, determine whether to use a SRFI or to rely on some kind of fallback, then we want the registry to only have the actually provided SRFIs.

2. If the registry is instead used only to index the available SRFIs names with numbers and to provide some clarity or glossary like information, without any implied or expected contract around provision of the implementation of said SRFIs, then it's fine and probably better to have them all there. 

I don't know enough about how people are using the registry system to make that call, but I'm happy with either one provided that the above criteria are met.

Reply:
Looking this over, it seems like choice 1 above is the implied behavior, in which case I'm inclined to keep the filtering, but someone correct me here. :-)

Reply:
I imagined it's choice 1, but I'm not sure either. OK, where does the registry come from? :D `registry.sls` says it was written by Derick Eddington - the guy who wrote SRFIs 103 & 104. But neither of those SRFIs says anything about a registry.

Reply:
Since the registry only exports feature lists -- `expand-time-features`, `run-time-features` and `available-features` -- that would strongly suggest that it's supposed to be about the available/implemented SRFIs and not all of them.

Reply:
Okay, well, I think it's easier to work off of the assumption that it should be filtered down, and then remove that later if it is really an issue, as it appears to me that it would be more difficult to add filtering in later than to remove it later.